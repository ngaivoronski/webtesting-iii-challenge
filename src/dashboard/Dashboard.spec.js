// Test away

import React from "react";
import Dashboard from "./Dashboard";
import { render } from "@testing-library/react";


test('Dashboard renders correctly',()=>{
    expect(render(<Dashboard/>)).toMatchSnapshot();
})

test('Shows controls and display', () => {
    const { getByTestId } = render(<Dashboard />);
    getByTestId(/display-component/i);
    getByTestId(/controls-component/i);
  });
