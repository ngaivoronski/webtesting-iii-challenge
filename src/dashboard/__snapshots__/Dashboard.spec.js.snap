// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="display panel"
        data-testid="display-component"
      >
        <div
          class="led green-led"
          data-testid="locked-div"
        >
          Unlocked
        </div>
        <div
          class="led green-led"
          data-testid="closed-div"
        >
          Open
        </div>
      </div>
      <div
        class="controls panel"
        data-testid="controls-component"
      >
        <button
          class="toggle-btn"
          data-testid="lock-button"
          disabled=""
        >
          Lock Gate
        </button>
        <button
          class="toggle-btn"
          data-testid="close-button"
        >
          Close Gate
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="display panel"
      data-testid="display-component"
    >
      <div
        class="led green-led"
        data-testid="locked-div"
      >
        Unlocked
      </div>
      <div
        class="led green-led"
        data-testid="closed-div"
      >
        Open
      </div>
    </div>
    <div
      class="controls panel"
      data-testid="controls-component"
    >
      <button
        class="toggle-btn"
        data-testid="lock-button"
        disabled=""
      >
        Lock Gate
      </button>
      <button
        class="toggle-btn"
        data-testid="close-button"
      >
        Close Gate
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
